client/ayon_core/hosts/max/plugins/publish/validate_model_name.py

```diff
@@ -0,0 +1,100 @@
+# -*- coding: utf-8 -*-
+"""Validate model nodes names."""
+import re
+
+import pyblish.api
+from pymxs import runtime as rt
+
+from ayon_core.hosts.max.api.action import SelectInvalidAction
+
+from ayon_core.pipeline.publish import (
+    OptionalPyblishPluginMixin,
+    PublishValidationError,
+    ValidateContentsOrder)
+
+
+class ValidateModelName(pyblish.api.InstancePlugin,
+                        OptionalPyblishPluginMixin):
+    """Validate Model Name
+    Validation regex is (.*)_(?P<subset>.*)_(GEO) by default.
+    e.g. {SOME_RANDOM_NAME}_{YOUR_SUBSET_NAME}_GEO should be your
+        default model name
+
+    The regex of (?P<subset>.*) can be replaced by (?P<asset>.*)
+    and (?P<project>.*).
+    e.g.
+    - (.*)_(?P<asset>.*)_(GEO) check if your model name is
+        {SOME_RANDOM_NAME}_{CURRENT_ASSET_NAME}_GEO
+    - (.*)_(?P<project>.*)_(GEO) check if your model name is
+        {SOME_RANDOM_NAME}_{CURRENT_PROJECT_NAME}_GEO
+
+    """
+    optional = True
+    order = ValidateContentsOrder
+    hosts = ["max"]
+    families = ["model"]
+    label = "Validate Model Name"
+    actions = [SelectInvalidAction]
+    regex = ""
+
+    @classmethod
+    def get_invalid(cls, instance):
+        invalid = []
+        model_names = [model.name for model in instance.data.get("members")]
+        cls.log.debug(model_names)
+        if not model_names:
+            cls.log.error("No Model found in the OP Data.")
+            invalid.append(model_names)
+        for name in model_names:
+            invalid_model_name = cls.get_invalid_model_name(instance, name)
+            invalid.extend(invalid_model_name)
+
+        return invalid
+
+    @classmethod
+    def get_invalid_model_name(cls, instance, name):
+        invalid = []
+        regex = cls.regex
+        reg = re.compile(regex)
+        matched_name = reg.match(name)
+        project_name = instance.context.data["projectName"]
+        current_asset_name = instance.context.data["folderPath"]
+        if matched_name is None:
+            cls.log.error("invalid model name on: {}".format(name))
+            cls.log.error("name doesn't match regex {}".format(regex))
+            invalid.append((rt.getNodeByName(name),
+                            "Model name doesn't match regex"))
+        else:
+            if "asset" in reg.groupindex:
+                if matched_name.group("asset") != current_asset_name:
+                    cls.log.error(
+                        "Invalid asset name of the model {}.".format(name)
+                    )
+                    invalid.append((rt.getNodeByName(name),
+                                    "Model with invalid asset name"))
+            if "subset" in reg.groupindex:
+                if matched_name.group("subset") != instance.name:
```

Comment:
`instance.name` is not by definition the subset name, often it is. But it isn't a hard requirement. You should really be checking against `instance.data["subset"]` here.

Also, this code would be much more readable if:

1. You do the node conversion once at the top `node = rt.getNodeByName(name)` instead of in all the invalid.append()` calls. Or even better instead of first pre-processing them to names, to keep the objects themselves around in the functions.

2. You do not include a message with the invalid append logic. It appears you are not even using the messages either?
3.

Reply:
Note also that you're checking against the context project name and folder path, even though the instance MAY be publishing to a different folder than current context. As such, you're then actually validating against the wrong folder path?